Add bullet to k8s access intro on deploying teleport within k8s (#11864)
gravitational/teleport: docs/pages/kubernetes-access/introduction.mdx

- Users can receive short-lived kubeconfig files (and certificates) using Single Sign-On (SSO) and switch between clusters without logging in twice.
- Admins can use roles to implement policies, such as the best practice that developers must not access production.
- Organizations can achieve compliance by capturing `kubectl` events and session recordings for `kubectl`.

> - Admins can deploy Teleport within a Kubernetes Cluster as a self-hosted Teleport deployment.

Reviewer: ```suggestion
- Admins can host the Teleport Auth Service and Proxy Service within a Kubernetes Cluster.
```
Suggested to reduce repetition
